

Checking out the cow samples


data split between paired and single strand sequencing

Processed samples are at 
'/grid/gillis/data/hover/work/cmxci1.paired/Bos_taurus'
'/grid/gillis/data/hover/work/cmxci1.single/Bos_taurus'

```{r}
library(ggplot2)
library(ggExtra)
library(stringr)
library(parallel)
library(dplyr)
```



```{r}

processed_files = list.files('/grid/gillis/data/hover/work/cmxci.paired/Bos_taurus')
wasp_vcf_files_paired = processed_files[grepl('.wasp.chrX.snps_filtered.vcf',processed_files  )]
#Ignore the .vcf.idx files
wasp_vcf_files_paired = wasp_vcf_files_paired[!grepl('.idx',wasp_vcf_files_paired )]
wasp_wig_files_paired = processed_files[grepl('.wasp.chrX.split.filtered.wig',processed_files  )]

length(wasp_vcf_files_paired)
length(wasp_wig_files_paired)

head(wasp_vcf_files_paired)
```


```{r}

processed_files = list.files('/grid/gillis/data/hover/work/cmxci.single/Bos_taurus')
wasp_vcf_files_single = processed_files[grepl('.wasp.chrX.snps_filtered.vcf',processed_files  )]
#Ignore the .vcf.idx files
wasp_vcf_files_single = wasp_vcf_files_single[!grepl('.idx',wasp_vcf_files_single )]
wasp_wig_files_single = processed_files[grepl('.wasp.chrX.split.filtered.wig',processed_files  )]

length(wasp_vcf_files_single)
length(wasp_wig_files_single)

head(wasp_vcf_files_single)
```

```{r}
wig_samples = sapply(strsplit(wasp_wig_files_paired, split = '.', fixed = T), '[[', 1)
wig_samples = c(wig_samples, sapply(strsplit(wasp_wig_files_single, split = '.', fixed = T), '[[', 1) ) #Add the single samples 
vcf_samples = sapply(strsplit(wasp_vcf_files_paired, split = '.', fixed = T), '[[', 1)
vcf_samples = c(vcf_samples, sapply(strsplit(wasp_vcf_files_single, split = '.', fixed = T), '[[', 1) )
table(wig_samples == vcf_samples)

```




```{r}


wasp_vcf_files_paired = paste('/grid/gillis/data/hover/work/cmxci.paired/Bos_taurus/',wasp_vcf_files_paired, sep = '')
wasp_vcf_files_single = paste('/grid/gillis/data/hover/work/cmxci.single/Bos_taurus/',wasp_vcf_files_single, sep = '')

wasp_wig_files_paired = paste('/grid/gillis/data/hover/work/cmxci.paired/Bos_taurus/',wasp_wig_files_paired, sep = '')
wasp_wig_files_single = paste('/grid/gillis/data/hover/work/cmxci.single/Bos_taurus/',wasp_wig_files_single, sep = '')


wasp_vcf_files = c(wasp_vcf_files_paired, wasp_vcf_files_single)
wasp_wig_files = c(wasp_wig_files_paired, wasp_wig_files_single)


length(wasp_vcf_files)

```

###########################
Screen for male samples using XIST and SRY expression from the STAR output
###########################

For cow, we used a female genome, so no SRY gene to map too. Start by looking at the distribution of XIST mapped reads



```{r}

star_output = paste(wig_samples, '.wasp.ReadsPerGene.out.tab', sep = '')
star_output = paste("/grid/gillis/data/hover/work/cmxci.paired/Bos_taurus/", star_output, sep = '')
star_output
```


```{r}

xist_reads = vector(mode = 'numeric', length = length(star_output))


for(i in 1:length(star_output)){
  
  temp_star_mapped_genes = read.table(star_output[i], skip = 4)  
  xist_reads[i] = temp_star_mapped_genes$V2[temp_star_mapped_genes$V1 == "XIST"]
}


hist(xist_reads)


```
```{r}
hist(xist_reads[xist_reads <= 2000], breaks = seq(0, 2000, 5))
potential_male = which(xist_reads == 0)

```




Function for merging the wig and vcf info, filtering for heterozygous SNPs with 2 alleles and a minimum of 10 reads per allele


```{r}

get_exp_vcf = function(vcf_file_path, wig_file_path){

  #Read in the vcf and wig files
  vcf = read.table(vcf_file_path)
  wig = read.table(wig_file_path, skip = 3)
  
  # Combine negative and positive strand counts
  wig_counts = wig[,1:6]        #test2=wig_counts
  wig_counts[,2:6] = wig_counts[,2:6] + wig[,9:13]
  colnames(wig_counts) = c( "Pos", "A", "C", "G", "T", "N" )
  
  #Get total read counts for each position
  tot_r_counts = rowSums(wig_counts[,2:6]) 
  #Delineate the number of SNPs at a position
  b = rowSums(wig_counts[,2:5]>0)
  
  # Check and keep positions with at least two SNPs exist
  f =  b >= 2
   
  # Calculate ratios and merge data
  d = wig_counts[f,2:5]/tot_r_counts[f]
  d[d==0]=NA 
  d = as.matrix(d)
  
  test3.mask = wig_counts[ f ,1:5]  #Positions and base counts
  test3.mask[test3.mask < 15] = 0       #Set base counts lees than 10 to 0, excluding these.  READ COUNT FILTER##########################################
  b3.mask = rowSums(test3.mask[,2:5]>0) #Filtered, delineate number of SNPS at a position
  
  #If there's no SNPs present, return NA
  if(sum(b3.mask == 2) <= 1){return(NA)}
  
  # Skip polymorphic sites (ie more than SNPs)
  m = match(test3.mask[b3.mask==2,1], vcf[,2])  #Match the positions that have only 2 alleles to their chromosome positions
  f.t = !is.na(m)                               #The ones that matched
  f.v = m[f.t] 
  
  temp.vcf = cbind(d[b3.mask==2,][f.t,], vcf[f.v,4:5]) #get the ratios of positions that only have 2 alleles, and add on the Ref and Alt allele from the VCF file

  #If there are no heterozygous SNPs with mapped reads on two alleles, skip
  if(length(m) <= 1){return(NA)}
  
  # Tidy up, find ref and alt
  if(dim(vcf[f.v,4:5])[1] <= 1){return(NA)}  #only one heterozygous SNP
  temp.vcf = cbind(d[b3.mask==2,][f.t,], vcf[f.v,4:5]) #get the ratios of positions that only have 2 alleles, and add on the Ref and Alt allele from the VCF file
  
  ref1 = as.character(temp.vcf[,5]) == "A"             #Grab all the reference A's, and so on
  ref2 = as.character(temp.vcf[,5]) == "C"
  ref3 = as.character(temp.vcf[,5]) == "G"
  ref4 = as.character(temp.vcf[,5]) == "T"
  
  alt1 = as.character(temp.vcf[,6]) == "A"             #Grab all the alternative A's, and so on
  alt2 = as.character(temp.vcf[,6]) == "C"
  alt3 = as.character(temp.vcf[,6]) == "G"
  alt4 = as.character(temp.vcf[,6]) == "T"
  
  
  ref.only = c(temp.vcf[ref1,1], temp.vcf[ref2,2], temp.vcf[ref3,3], temp.vcf[ref4,4])  #Combine all the reference SNPs ratios
  alt.only = c(temp.vcf[alt1,1], temp.vcf[alt2,2], temp.vcf[alt3,3], temp.vcf[alt4,4])  #Combine all the alternative SNPs ratios
  
  
  temp2.vcf = cbind(test3.mask[b3.mask==2,2:5][f.t,], vcf[f.v,4:5])                     #Grab the filtered SNPs and their alleles, for only those with two alleles
  temp3.vcf = temp2.vcf[,1:2]                                                           #Set up structure to hold the combined Ref / Alt counts per position of the filtered SNPS
  temp3.vcf = temp3.vcf*0
  colnames(temp3.vcf) = c('ref_counts','alt_counts')
  
  
  #Starting filling in the ref and alt counts
  #Produces the ref and alt read counts for the heterozygous SNPs
  temp3.vcf[ref1,1] = temp2.vcf[ref1,1]
  temp3.vcf[ref2,1] = temp2.vcf[ref2,2]
  temp3.vcf[ref3,1] = temp2.vcf[ref3,3]
  temp3.vcf[ref4,1] = temp2.vcf[ref4,4]
  
  temp3.vcf[alt1,2] = temp2.vcf[alt1,1]
  temp3.vcf[alt2,2] = temp2.vcf[alt2,2]
  temp3.vcf[alt3,2] = temp2.vcf[alt3,3]
  temp3.vcf[alt4,2] = temp2.vcf[alt4,4]
  
  
  #filtering out Variants with , in them, they are multiple alleles missed previously
  
  allele.f1 = sapply(1:length(temp2.vcf$V4),function(i) str_count(as.character(temp2.vcf$V4)[i], ',') )    #Ref alleles with commas
  allele.f2 = sapply(1:length(temp2.vcf$V5),function(i) str_count(as.character(temp2.vcf$V5)[i], ',') )    #Alt alleles with commas
  allele.f = !(allele.f1 == 1 | allele.f2 == 1)                                                            #Get rid of them
  
  temp2.vcf = temp2.vcf[allele.f, ]
  temp3.vcf = temp3.vcf[allele.f, ]
  
  vcf_2 = cbind(test3.mask[b3.mask==2,1][f.t][allele.f], temp3.vcf, temp2.vcf)
  colnames(vcf_2) = c('Pos','ref_counts','alt_counts','A','C','G','T', 'ref','alt')
  
  return(vcf_2)
   
}
  
  
```


```{r}
Sys.time()
wasp_filt_vcf = mclapply(1:length(wasp_vcf_files), function(i) get_exp_vcf(vcf_file_path = wasp_vcf_files[i], 
                                                                           wig_file_path = wasp_wig_files[i]), mc.cores = 5)
Sys.time()

```


```{r}

head(wasp_filt_vcf)


```


Number of filtered SNPs kept
```{r}

num_snps = vector(mode = 'numeric', length = length(wasp_filt_vcf))

for(i in 1:length(wasp_filt_vcf)){
  data = wasp_filt_vcf[[i]]
  
  if(is.null(dim(data))){
    num_snps[i] = 0
  }else{num_snps[i] = dim(data)[1]}
}

hist(num_snps, breaks = seq(0,max(num_snps), 2))


```

```{r}

max(num_snps)
hist(num_snps, breaks = seq(0,max(num_snps)+10, 6))

```

```{r}
table(num_snps >= 300)

high_snp_index = which(num_snps >= 300)

for(i in high_snp_index){
  
  data = wasp_filt_vcf[[i]]
  tot_counts = data$ref_counts + data$alt_counts
  ref_ratios = data$ref_counts / tot_counts
  
  hist(ref_ratios, breaks = seq(0,1,.01))
  
}



```

Get the gene specific distributions


Getting the genome annotations
Genomes are currently located on Elzar at /grid/gillis/data_norepl/hover/genomes
Copied over a few to rugen18

```{r}


cow_gtf = rtracklayer::import('/home/werner/projects/cross_species_XCI/genomes/Bos_taurus/annotation.gtf')
cow_gtf = as.data.frame(cow_gtf)

#Genome specific x chromosome name
cow_x_chrom_name = readLines('/home/werner/projects/cross_species_XCI/genomes/Bos_taurus/chrXlabel.txt', n = 1)

cow_x_gtf = cow_gtf[cow_gtf$seqnames == cow_x_chrom_name, ]

cow_x_genenames = unique(cow_x_gtf$gene)
length(cow_x_genenames)
```


Adding gene annotations to each sample

```{r}
test = wasp_filt_vcf[[1]]
test

```

```{r}
cow_x_meta = cow_x_gtf[cow_x_gtf$gbkey == 'Gene',]
cow_x_meta

```




```{r}
cow_x_meta = cow_x_gtf[cow_x_gtf$gbkey == 'Gene',]
x_gene_starts = cow_x_meta$start
x_gene_ends = cow_x_meta$end


annot_cow_vcfs = wasp_filt_vcf

for(j in 1:length(annot_cow_vcfs)){
  test = annot_cow_vcfs[[j]]
  if(is.null(dim(test))){next}
  
  
  gene_annots = vector(mode = 'character', length = length(test$Pos))
  gene_biotype = vector(mode = 'character', length = length(test$Pos))
  for( i in 1:dim(test)[1]){
  
    test_gene_start = test$Pos[i]
    gene_index = findInterval(test_gene_start, x_gene_starts)
    if(gene_index == 0){next}
    
    if(test_gene_start <= x_gene_ends[gene_index]){
      
      gene_annots[i] = cow_x_meta$gene[gene_index]
      gene_biotype[i] = cow_x_meta$gene_biotype[gene_index]
    }
  }
  
  test$gene = gene_annots
  test$gene_biotype = gene_biotype
  annot_cow_vcfs[[j]] = test
}

head(annot_cow_vcfs)

```


save the annotated vcfs with all the SNPs present


```{r}
save(annot_cow_vcfs, file = '/home/werner/projects/cross_species_XCI/data/species_data_analysis/bos_taurus/all_snp_vcf_readFilt_diff.Rdata')


```

```{r}
load('/home/werner/projects/cross_species_XCI/data/species_data_analysis/bos_taurus/all_snp_vcf_readFilt_diff.Rdata')

```


```{r}
annot_cow_vcfs[1:5]

```




Looking at gene specific expression 


```{r}

captured_genes = c()
for( i in 1:length(annot_cow_vcfs)){
  suppressWarnings(if(is.null(dim(annot_cow_vcfs[[i]]))){next})
  captured_genes = c(captured_genes, annot_cow_vcfs[[i]]$gene)
}

captured_genes = unique(captured_genes)
captured_genes = captured_genes[captured_genes != ""]

length(captured_genes)
```


```{r}
for(i in 1:100){
  
  current_gene = captured_genes[i]
  
  gene_ref_exp = c()
  for(j in 1:length(annot_cow_vcfs)){
    suppressWarnings(if(is.null(dim(annot_cow_vcfs[[j]]))){next})
    gene_index = annot_cow_vcfs[[j]]$gene == current_gene
    ref_counts = annot_cow_vcfs[[j]]$ref_counts[gene_index]
    alt_counts = annot_cow_vcfs[[j]]$alt_counts[gene_index]
    total_counts = ref_counts + alt_counts
    gene_ref_exp = c(gene_ref_exp, ref_counts / total_counts )

  }
  
  
  hist(gene_ref_exp, main = current_gene, xlim = c(0,1), breaks = seq(0,1, .025))  
  
}

```



Checking SNP specific ratio distributions, seems to be only certain SNPs that are causing problems

```{r}

captured_SNPs = c()
for( i in 1:length(annot_cow_vcfs)){
  suppressWarnings(if(is.null(dim(annot_cow_vcfs[[i]]))){next})
  captured_SNPs = c(captured_SNPs, annot_cow_vcfs[[i]]$Pos)
}

captured_SNPs = unique(captured_SNPs)
captured_SNPs = captured_SNPs[captured_SNPs != ""]

length(captured_SNPs)
```
Grab the mean read counts and variance in read counts as well

```{r}

snp_stat_df = data.frame(snp_pos = integer(length(captured_SNPs)), num_samples_present = integer(length(captured_SNPs)), 
                         mean_ref_ratio = integer(length(captured_SNPs)), mean_total_counts = integer(length(captured_SNPs)), sd_total_counts = integer(length(captured_SNPs)))



for(i in 1:length(captured_SNPs)){

  if(i %% 1000 == 0){print(i)}
  current_SNP = captured_SNPs[i]
  
  snp_ref_exp = c()
  snp_total_reads = c()
  for(j in 1:length(annot_cow_vcfs)){
    suppressWarnings(if(is.null(dim(annot_cow_vcfs[[j]]))){next})
    gene_index = annot_cow_vcfs[[j]]$Pos == current_SNP
    ref_counts = annot_cow_vcfs[[j]]$ref_counts[gene_index]
    alt_counts = annot_cow_vcfs[[j]]$alt_counts[gene_index]
    total_counts = ref_counts + alt_counts
    snp_ref_exp = c(snp_ref_exp, ref_counts / total_counts )
    snp_total_reads = c(snp_total_reads, total_counts)
    
    
  }
  
  snp_stat_df$snp_pos[i] = current_SNP
  snp_stat_df$num_samples_present[i] = length(snp_ref_exp)
  snp_stat_df$mean_ref_ratio[i] = mean(snp_ref_exp)
  snp_stat_df$mean_total_counts[i] = mean(snp_total_reads)
  snp_stat_df$sd_total_counts[i] = sd(snp_total_reads)
}

```


```{r}
hist(snp_stat_df$mean_ref_ratio, breaks = seq(0,1,.01))
hist(snp_stat_df$num_samples_present)
par(pty = 's')
plot(snp_stat_df$mean_ref_ratio,snp_stat_df$num_samples_present, cex = .25, xlab = 'Mean reference ratio', 
     ylab = '# of samples SNP is detected', main = 'Bos taurus Het SNPs')
abline(v = .5, col = 'red')
abline(h = 2100, col = 'blue')
abline(v = .4, col = 'blue')
abline(v = .6, col = 'blue')
```


```{r}

plot(snp_stat_df$mean_total_counts, snp_stat_df$sd_total_counts, cex = .5)

```


Filter out SNPs present in at least 20 samples that have a mean reference ratio < .4 or > .6, 
Should get rid of SNPs with consistent reference bias without excluding whole genes



```{r}

#index_4 = snp_stat_df$num_samples_present >= 10 & snp_stat_df$mean_ref_ratio < .4
#index_6 = snp_stat_df$num_samples_present >= 10 & snp_stat_df$mean_ref_ratio > .6

#total_index = index_4 | index_6
#table(index_4)
#table(index_6)
#table(total_index)

#snp_stat_df$filter = !total_index

snp_stat_df$filter = !(snp_stat_df$mean_ref_ratio < .4 | snp_stat_df$mean_ref_ratio > .6)
table(snp_stat_df$filter)

```


Save the SNP dataframe

```{r}

save(snp_stat_df, file = '/home/werner/projects/cross_species_XCI/data/species_data_analysis/bos_taurus/detected_het_snps_df_readFilt_diff.Rdata')

```

```{r}
load('/home/werner/projects/cross_species_XCI/data/species_data_analysis/bos_taurus/detected_het_snps_df_readFilt_diff.Rdata')

```

```{r}
snp_stat_df

```



Filtering the vcfs for only annotated genes and the max powered SNP per gene


```{r}
filt_cow_vcfs = annot_cow_vcfs

for(i in 1:length(filt_cow_vcfs)){
  test = filt_cow_vcfs[[i]]
  
  suppressWarnings(if(is.null(dim(test))){next})
  
  #Get rid of SNPs in nonannotated genes
  test = test[test$gene != "", ]
  suppressWarnings(if(dim(test)[1] == 0){filt_cow_vcfs[[i]] = NA; next})

  #Get the unique number of genes
  sample_genes = unique(test$gene)
  test_2 = data.frame(matrix(NA, 
                             nrow = length(sample_genes), 
                             ncol = dim(test)[2]))
  colnames(test_2) = colnames(test)
  
  
  for(j in 1:length(sample_genes)){
    
    current_gene_data = test[test$gene == sample_genes[j], ]
    if(dim(current_gene_data)[1] == 1){
      test_2[j, ] = current_gene_data
    }else{
      
      current_tot_exp = current_gene_data$ref_counts + current_gene_data$alt_counts
      test_2[j, ] = current_gene_data[which.max(current_tot_exp), ]
    }
  }
  #Only keep SNPs with a total of at least 10 reads
  total_counts = test_2$ref_counts + test_2$alt_counts
  test_2 = test_2[total_counts >= 30, ]
  suppressWarnings(if(dim(test_2)[1] == 0){filt_cow_vcfs[[i]] = NA; next})
  filt_cow_vcfs[[i]] = test_2
}

```

only keep SNPs with a total of 10 reads total








save the filtered vcf list

```{r}
save(filt_cow_vcfs, file = '/home/werner/projects/cross_species_XCI/data/species_data_analysis/bos_taurus/max_snp_vcf_readFilt_diff.Rdata')


```

```{r}
load('/home/werner/projects/cross_species_XCI/data/species_data_analysis/bos_taurus/max_snp_vcf_readFilt_diff.Rdata')

```


```{r}

filt_cow_vcfs[1:5]

```



Number of genes per samples


```{r}

num_filt_genes = vector(mode = 'numeric', length = length(filt_cow_vcfs))
for(i in 1:length(filt_cow_vcfs)){
  
  suppressWarnings(if(is.null(dim(filt_cow_vcfs[[i]]))){
    num_filt_genes[i] = 0
    next
    
  })
  
  num_filt_genes[i] = dim(filt_cow_vcfs[[i]])[1]

}



```

```{r}

hist(num_filt_genes, breaks = seq(0,300,2))

sum(num_filt_genes >= 10)
```

look at the SNP distributions, with and without including the potentially reference biased SNPs

```{r}

index_num_samps = which(num_filt_genes >= 30)

```


```{r}

index = seq(200,300,1)

for(i in index){

  data = filt_cow_vcfs[[index_num_samps[i]]]
  
  snp_filter = data$Pos %in% snp_stat_df$snp_pos[snp_stat_df$filter]
  
  total_exp = data$ref_counts + data$alt_counts
  ref_ratios = data$ref_counts[snp_filter] / total_exp[snp_filter]
  
  #ref_ratios = data$ref_counts / total_exp
  
  hist(ref_ratios, xlim = c(0,1), breaks = seq(0,1, .01))
  abline(v = .5, col = 'red')

}
```


first pass at skew modeling. Filter out the biased SNPs, there's only a handful of potential male samples with enough het SNPs, which makes sense, but also not a large concern at the moment at least for the cows




Skew modeling



```{r}


# Functions to fold and unfold ratios 
folded <- function(x) { apply( cbind(x,1-x), 1, max)} 
unfold <- function(x) { c(x,1-x) } 

#Sara's MLE function
# Maximum likelihood estimate function. Note, x is a global variable! Should fix but this lets me use it in the bootstrap later.  
mle_folded <- function(x){ 
  mus = seq(0.5,1, by = 0.001)
  sigmas = seq(0,0.5, by = 0.01)
  
  mles = sapply(mus, function(mu)
    sapply(sigmas, function(sigma)
      -sum( log(dnorm( x,   mean = mu, sd = sigma )
                + dnorm( x,   mean = 1-mu, sd = sigma ) )  )))
  mles[!is.finite(mles)] = NA
  coefs = which(mles==min(mles, na.rm=T), arr.ind=T)
  return (list( mus[coefs[2]] , sigmas[coefs[1]]))
} 

```



```{r}
Ns = vector(mode='numeric', length = length(filt_cow_vcfs))

for(i in 1:length(filt_cow_vcfs)){
  
  suppressWarnings(if(is.null(dim(filt_cow_vcfs[[i]]))){Ns[i] = 0}
  else{Ns[i] = dim(filt_cow_vcfs[[i]])[1]})
  
}

```

```{r}

ratios.max = list()
for(i in 1: length(filt_cow_vcfs)){
  
  if(Ns[i] == 0){next}
  
  total_exp = filt_cow_vcfs[[i]]$ref_counts + filt_cow_vcfs[[i]]$alt_counts
  ref_ratios = filt_cow_vcfs[[i]]$ref_counts / total_exp
  ratios.max[[i]] = ref_ratios

}

head(ratios.max)

```


```{r}

est_skew_func = function(list.skew.max, Ns, ratios.max, snp_stat_df, filter_snps = T){
  
  #If there's only 1 or 0 SNPs, don't estimate a skew
  if(Ns <= 0){return(NA)}
  
  #Filter for the good snps
  if(filter_snps){
    snp_index = list.skew.max$Pos %in% snp_stat_df$snp_pos[snp_stat_df$filter]
    #Fold the reference skews
    folded_ref_skews = folded(ratios.max[snp_index])
  }else{
    folded_ref_skews = folded(ratios.max)
  }
  #return the mean and sigma estimates
  return(mle_folded(folded_ref_skews))
}

```


```{r}

Sys.time()
folded_norm_fits = mclapply(1:length(filt_cow_vcfs), function(i) est_skew_func(filt_cow_vcfs[[i]], Ns[i], ratios.max[[i]], snp_stat_df), mc.cores = 20)
Sys.time()


Sys.time()
folded_norm_fits_bad_snps = mclapply(1:length(filt_cow_vcfs), function(i) est_skew_func(filt_cow_vcfs[[i]], Ns[i], ratios.max[[i]], snp_stat_df, filter_snps = F), mc.cores = 20)
Sys.time()

#Grab the skew and variance estimates
est_skew = vector(mode = 'numeric', length = length(folded_norm_fits))
est_var = vector(mode = 'numeric', length = length(folded_norm_fits))
for(i in 1:length(folded_norm_fits)){
  
  if(is.na(folded_norm_fits[i])){
    est_skew[i] = NA
    est_var[i] = NA
    next
  }
  
  est_skew[i] = folded_norm_fits[[i]][[1]]
  est_var[i] = folded_norm_fits[[i]][[2]]
}

#Grab the skew and variance estimates
est_skew_bad_snps = vector(mode = 'numeric', length = length(folded_norm_fits_bad_snps))
est_var_bad_snps = vector(mode = 'numeric', length = length(folded_norm_fits_bad_snps))
for(i in 1:length(folded_norm_fits_bad_snps)){
  
  if(is.na(folded_norm_fits_bad_snps[i])){
    est_skew_bad_snps[i] = NA
    est_var_bad_snps[i] = NA
    next
  }
  
  est_skew_bad_snps[i] = folded_norm_fits_bad_snps[[i]][[1]]
  est_var_bad_snps[i] = folded_norm_fits_bad_snps[[i]][[2]]
}

```

```{r}

hist(est_skew, main = 'Cow skew estimates filtering out bad SNPs', breaks=64)
hist(est_var, main = 'Variance estimates', breaks = 32)
plot(est_skew, est_var)



```


```{r}

hist(est_skew, main = 'Cow skew estimates filtering out bad SNPs', breaks=64)
hist(est_var, main = 'Variance estimates', breaks = 32)
plot(est_skew, est_var)



```

```{r}
par(pty = 's')
plot(est_skew, est_skew_bad_snps, cex = .5)
abline(a = 0, b = 1, col = 'red')
```



Make a dataframe from with the estimated skew stats

```{r}

num_good_snps = vector(length = length(filt_cow_vcfs), mode = 'numeric')
total_snps = vector(length = length(filt_cow_vcfs), mode = 'numeric')

for(i in 1:length(filt_cow_vcfs)){
  
  data = filt_cow_vcfs[[i]]
  suppressWarnings(if(is.null(dim(data))){num_good_snps[i] = 0; total_snps[i] = 0; next})
  
  snp_filter = data$Pos %in% snp_stat_df$snp_pos[snp_stat_df$filter]
  
  total_snps[i] = length(data$Pos)
  num_good_snps[i] = length(data$Pos[snp_filter])
}

plot(total_snps, num_good_snps)

```

######################################
Trying to find escape genes

######################################


```{r}

all_genes = c()

for(i in 1:length(filt_cow_vcfs)){
  if(is.null(dim(filt_cow_vcfs[[i]]))){next}
  all_genes = c(all_genes,filt_cow_vcfs[[i]]$gene )
}

all_genes = unique(all_genes)
length(all_genes)

```

Get a p-value per sample for gene's deviation from .5 in skewed samples. Small p-value is balanced biallelic expression in skewed samples
Fishers method to combine p values over all the samples

```{r}

gene_chi_right_p_values = vector(mode = 'numeric', length = length(all_genes))
names(gene_chi_right_p_values) = all_genes
gene_chi_left_p_values = vector(mode = 'numeric', length = length(all_genes))
names(gene_chi_left_p_values) = all_genes
num_pvalues =  vector(mode = 'numeric', length = length(all_genes))
skewed_index = which(est_skew> .6)

for(j in 1:length(all_genes)){

  p_value_vec = c()
  num_genes_vec = c()
  for(i in 1:length(skewed_index)){
  
    if(is.null(dim(filt_cow_vcfs[[skewed_index[i]]]))){next}
    if(!all_genes[j] %in% filt_cow_vcfs[[skewed_index[i]]]$gene){next}
    
    gene_index = filt_cow_vcfs[[skewed_index[i]]]$gene == all_genes[j]
    
    ratios = filt_cow_vcfs[[skewed_index[i]]]$ref_counts / (filt_cow_vcfs[[skewed_index[i]]]$ref_counts + filt_cow_vcfs[[skewed_index[i]]]$alt_counts)
    deviations = abs(ratios - .5)
    gene_deviation = deviations[gene_index]
    deviations = deviations[!gene_index]
    p_value = sum(deviations <= gene_deviation) / length(deviations)
    num_genes = length(deviations)
    p_value_vec = c(p_value_vec, p_value)
    num_genes_vec = c(num_genes_vec, num_genes)
  
  }
  
  if(length(p_value_vec) == 0){next}
  p_value_vec[p_value_vec == 0] = 1e-5
  hist(p_value_vec[num_genes_vec >= 30], breaks = seq(0,1,.025), main = all_genes[j])
  
  #Get the Fishers method p-value for this gene
  chi_stat = -2 * sum(log(p_value_vec[num_genes_vec >= 30]))
  gene_chi_right_p_values[j] = pchisq(chi_stat, 2*sum(num_genes_vec >= 30), lower.tail = F)
  gene_chi_left_p_values[j] = pchisq(chi_stat, 2*sum(num_genes_vec >= 30), lower.tail = T)
  num_pvalues[j] = sum(num_genes_vec >= 30)
     
}

```

```{r}

adj_gene_chi_right_p_values = p.adjust(gene_chi_right_p_values, method = 'BH')
adj_gene_chi_left_p_values = p.adjust(gene_chi_left_p_values, method = 'BH')


hist(adj_gene_chi_right_p_values, breaks = seq(0,1,.025))
hist(adj_gene_chi_right_p_values[num_pvalues >= 30], breaks = seq(0,1,.025))

escape_gene_index = which(adj_gene_chi_right_p_values <= .001 & num_pvalues >=30)
adj_gene_chi_right_p_values[escape_gene_index]
all_genes[escape_gene_index]


```
```{r}

adj_gene_chi_left_p_values = p.adjust(gene_chi_left_p_values, method = 'BH')
adj_gene_chi_left_p_values = p.adjust(gene_chi_left_p_values, method = 'BH')


hist(adj_gene_chi_left_p_values, breaks = seq(0,1,.025))
hist(adj_gene_chi_left_p_values[num_pvalues >= 30], breaks = seq(0,1,.025))

imprint_gene_index = which(adj_gene_chi_left_p_values <= .001 & num_pvalues >=30)
adj_gene_chi_left_p_values[imprint_gene_index]
all_genes[imprint_gene_index]


```
p-value distributions of genes called as escape

```{r}

skewed_index = which(est_skew> .6)

for(j in 1:length(all_genes[escape_gene_index])){

  gene = all_genes[escape_gene_index][j]
  
  p_value_vec = c()
  num_genes_vec = c()
  for(i in 1:length(skewed_index)){
  
    if(is.null(dim(filt_cow_vcfs[[skewed_index[i]]]))){next}
    if(!gene %in% filt_cow_vcfs[[skewed_index[i]]]$gene){next}
    
    gene_index = filt_cow_vcfs[[skewed_index[i]]]$gene == gene
    
    ratios = filt_cow_vcfs[[skewed_index[i]]]$ref_counts / (filt_cow_vcfs[[skewed_index[i]]]$ref_counts + filt_cow_vcfs[[skewed_index[i]]]$alt_counts)
    deviations = abs(ratios - .5)
    gene_deviation = deviations[gene_index]
    deviations = deviations[!gene_index]
    p_value = sum(deviations <= gene_deviation) / length(deviations)
    num_genes = length(deviations)
    
    p_value_vec = c(p_value_vec, p_value)
    num_genes_vec = c(num_genes_vec, num_genes)
  
  }
  
  p_value_vec[p_value_vec == 0] = 1e-5
  hist(p_value_vec[num_genes_vec >= 30], breaks = seq(0,1,.025), main =gene)
  
    
     
}

```

p-value distributions of genes preferentially always highly skewed
```{r}

skewed_index = which(est_skew> .6)

for(j in 1:length(all_genes[imprint_gene_index])){

  gene = all_genes[imprint_gene_index][j]
  
  p_value_vec = c()
  num_genes_vec = c()
  for(i in 1:length(skewed_index)){
  
    if(is.null(dim(filt_cow_vcfs[[skewed_index[i]]]))){next}
    if(!gene %in% filt_cow_vcfs[[skewed_index[i]]]$gene){next}
    
    gene_index = filt_cow_vcfs[[skewed_index[i]]]$gene == gene
    
    ratios = filt_cow_vcfs[[skewed_index[i]]]$ref_counts / (filt_cow_vcfs[[skewed_index[i]]]$ref_counts + filt_cow_vcfs[[skewed_index[i]]]$alt_counts)
    deviations = abs(ratios - .5)
    gene_deviation = deviations[gene_index]
    deviations = deviations[!gene_index]
    p_value = sum(deviations <= gene_deviation) / length(deviations)
    num_genes = length(deviations)
    
    p_value_vec = c(p_value_vec, p_value)
    num_genes_vec = c(num_genes_vec, num_genes)
  
  }
  
  p_value_vec[p_value_vec == 0] = 1e-5
  hist(p_value_vec[num_genes_vec >= 30], breaks = seq(0,1,.025), main =gene)
  
    
     
}

```



look at the expression levels of the SNPs from the escape genes and the imprint genes


```{r}


escape_genes = all_genes[adj_gene_chi_right_p_values <= .001 & num_pvalues >=30]
imprint_genes = all_genes[adj_gene_chi_left_p_values <= .001 & num_pvalues >=30]
escape_pos = c()
imprint_pos = c()
for(i in 1:length(filt_cow_vcfs)){
  if(is.null(dim(filt_cow_vcfs[[i]]))){next}
  data = filt_cow_vcfs[[i]]

  escape_pos = c(escape_pos, data$Pos[data$gene %in% escape_genes])
  imprint_pos = c(imprint_pos, data$Pos[data$gene %in% imprint_genes])

}

escape_pos = unique(escape_pos)
imprint_pos = unique(imprint_pos)

temp_snp_exp_df = snp_stat_df[ ,c('snp_pos','mean_total_counts')]
temp_snp_exp_df$label = rep('regular', length = nrow(temp_snp_exp_df))
temp_snp_exp_df$label[temp_snp_exp_df$snp_pos %in% escape_pos] = 'escape'
temp_snp_exp_df$label[temp_snp_exp_df$snp_pos %in% imprint_pos] = 'imprint'
table(temp_snp_exp_df$label)

temp_snp_exp_df$label = factor(temp_snp_exp_df$label, levels = c('regular','escape','imprint'))
ggplot(temp_snp_exp_df, aes(x = label, y = log10(mean_total_counts), color = label)) + geom_violin(scale = 'width', color = 'black') + geom_boxplot(width = .5) +
  scale_color_manual(values = c('regular' = 'darkgreen', 'escape' = 'red', 'imprint' = 'blue'))


```


pull together a dataframe of the escape stats per gene to save 

```{r}

escape_label = rep('non-escape', length = length(all_genes))
escape_label[adj_gene_chi_right_p_values <= .001 & num_pvalues >=30] = 'escape'
escape_cow_df = data.frame(gene = all_genes, num_pvalues = num_pvalues,
                           chi_right_p_values = gene_chi_right_p_values, chi_left_p_values = gene_chi_left_p_values,
                           adj_chi_right_p_values = adj_gene_chi_right_p_values, adj_chi_left_p_values = adj_gene_chi_left_p_values,
                           escape_label = escape_label)

escape_cow_df

save(escape_cow_df, file = '/home/werner/projects/cross_species_XCI/data/species_data_analysis/bos_taurus/escape_cow_df_readFilt_diff.Rdata')



```
##############################################
look at the chromosomal position of the escape genes
##############################################


```{r}
#same gene metadata as before
cow_x_meta = cow_x_gtf[cow_x_gtf$gbkey == 'Gene'& !is.na(cow_x_gtf$gbkey),]
x_gene_starts = cow_x_meta$start
x_gene_ends = cow_x_meta$end


escape_genes = escape_cow_df$gene[escape_cow_df$escape_label == 'escape']

#Get the mean expression of each gene. Take the mean of the mean total counts of the SNPs within that gene
gene_name_vec = vector(mode = 'character', length = nrow(snp_stat_df))
for(i in 1:nrow(snp_stat_df)){
  gene_index = findInterval(snp_stat_df$snp_pos[i], cow_x_meta$start)
  if(gene_index == 0){gene_name_vec[i] = NA; next}
  
  if(snp_stat_df$snp_pos[i] <= cow_x_meta$end[gene_index]){
    gene_name_vec[i] = cow_x_meta$gene[gene_index]
  }else{gene_name_vec[i] = NA}
}

#Add gene names to the snp stat df
snp_stat_df$gene = gene_name_vec
#Ignore SNPs not in annotated genes
temp_snp_stat_df = snp_stat_df[!is.na(snp_stat_df$gene), ]
mean_gene_df = temp_snp_stat_df %>% group_by(gene) %>% summarize(mean_exp = mean(mean_total_counts), num_snps = length(mean_total_counts))
mean_gene_df

gene_pos_df = data.frame(gene_name = cow_x_meta$gene, gene_start = cow_x_meta$start, gene_end = cow_x_meta$end, 
                         gene_biotype = cow_x_meta$gene_biotype, escape_label = cow_x_meta$gene %in% escape_genes)

index = match(gene_pos_df$gene_name, mean_gene_df$gene)
gene_pos_df$mean_exp = mean_gene_df$mean_exp[index]
gene_pos_df$num_snps = mean_gene_df$num_snps[index]

gene_pos_df

ggplot(gene_pos_df, aes(x = gene_start, y = log10(mean_exp), color = escape_label)) + geom_point(aes(size = escape_label)) +
  scale_color_manual(values = c('TRUE' = 'red', 'FALSE' = 'black')) +
  scale_size_manual(values = c('TRUE' = 2, 'FALSE' = 1))
ggplot(gene_pos_df, aes(x = gene_start, y = log10(num_snps), color = escape_label)) + geom_point(aes(size = escape_label)) +
  scale_color_manual(values = c('TRUE' = 'red', 'FALSE' = 'black')) +
  scale_size_manual(values = c('TRUE' = 2, 'FALSE' = 1))
```


################################
Get skew estimates excluding these potential escape genes

################################


save the vcfs without the escape genes

```{r}
filt_cow_vcfs_noEscape = filt_cow_vcfs
num_snp_escape = vector(mode = 'numeric', length = length(filt_cow_vcfs_noEscape))
for(i in 1:length(filt_cow_vcfs_noEscape)){
  
  if(is.null(dim(filt_cow_vcfs_noEscape[[i]]))){next}
  escape_index = filt_cow_vcfs_noEscape[[i]]$gene %in% escape_genes
  filt_cow_vcfs_noEscape[[i]] = filt_cow_vcfs_noEscape[[i]][!escape_index, ]
  num_snp_escape[i] = dim(filt_cow_vcfs_noEscape[[i]])
}


save(filt_cow_vcfs_noEscape, file = '/home/werner/projects/cross_species_XCI/data/species_data_analysis/bos_taurus/max_snp_vcf_noEscape_readFilt_diff.Rdata')

```

```{r}
load(file = '/home/werner/projects/cross_species_XCI/data/species_data_analysis/bos_taurus/max_snp_vcf_noEscape_readFilt_diff.Rdata')
filt_cow_vcfs_noEscape[1:5]
```


```{r}
Ns.escape = vector(mode='numeric', length = length(filt_cow_vcfs_noEscape))

for(i in 1:length(filt_cow_vcfs_noEscape)){
  
  suppressWarnings(if(is.null(dim(filt_cow_vcfs_noEscape[[i]]))){Ns.escape[i] = 0}
  else{Ns.escape[i] = dim(filt_cow_vcfs_noEscape[[i]])[1]})
  
}

```

```{r}

ratios.escape.max = list()
for(i in 1: length(filt_cow_vcfs_noEscape)){
  
  if(Ns.escape[i] == 0){next}
  
  total_exp = filt_cow_vcfs_noEscape[[i]]$ref_counts + filt_cow_vcfs_noEscape[[i]]$alt_counts
  ref_ratios = filt_cow_vcfs_noEscape[[i]]$ref_counts / total_exp
  ratios.escape.max[[i]] = ref_ratios

}

head(ratios.escape.max)

```


```{r}

Sys.time()
folded_norm_fits_no_escape = mclapply(1:length(filt_cow_vcfs_noEscape), function(i) 
  est_skew_func(filt_cow_vcfs_noEscape[[i]], Ns.escape[i], ratios.escape.max[[i]], snp_stat_df), mc.cores = 20)
Sys.time()

#Grab the skew and variance estimates
est_skew_no_escape = vector(mode = 'numeric', length = length(folded_norm_fits_no_escape))
est_var_no_escape = vector(mode = 'numeric', length = length(folded_norm_fits_no_escape))
for(i in 1:length(folded_norm_fits)){
  
  if(is.na(folded_norm_fits_no_escape[i])){
    est_skew_no_escape[i] = NA
    est_var_no_escape[i] = NA
    next
  }
  
  est_skew_no_escape[i] = folded_norm_fits_no_escape[[i]][[1]]
  est_var_no_escape[i] = folded_norm_fits_no_escape[[i]][[2]]
}


```



Compare skew estimates with and without escape

first get the number of SNPs per sample now excluding escape
```{r}

num_good_snps_escape = vector(length = length(filt_cow_vcfs_noEscape), mode = 'numeric')

for(i in 1:length(filt_cow_vcfs_noEscape)){
  
  data = filt_cow_vcfs_noEscape[[i]]
  suppressWarnings(if(is.null(dim(data))){num_good_snps_escape[i] = 0; next})
  
  snp_filter = data$Pos %in% snp_stat_df$snp_pos[snp_stat_df$filter]
  num_good_snps_escape[i] = length(data$Pos[snp_filter])
}

hist(num_good_snps_escape)

```


```{r}
hist(est_skew_no_escape[num_good_snps_escape >= 10], breaks = seq(.5, 1, .005))
hist(est_skew[num_good_snps >= 10], breaks = seq(.5, 1, .005))

par(pty = 's')
plot(est_skew, est_skew_no_escape, xlim = c(.5, 1), ylim = c(.5, 1))
abline(a = 0, b = 1, col = 'red')
```


```{r}

quantile(est_var_no_escape, probs = seq(0,1,.10), na.rm = T)


sample_index = which( num_good_snps_escape >= 25 & est_var_no_escape < .13)
length(sample_index)

for(i in 1:length(sample_index)){
  
  data = filt_cow_vcfs_noEscape[[sample_index[i]]]
  ref_ratios = data$ref_counts / (data$ref_counts + data$alt_counts)
  snp_filter = data$Pos %in% snp_stat_df$snp_pos[snp_stat_df$filter == TRUE]
  hist(ref_ratios[snp_filter], breaks = seq(0,1,.025))
  
  
}

```



```{r}
sample_index = num_good_snps_escape >= 10 & est_var_no_escape < .12 
sum(sample_index)
hist(unfold(est_skew_no_escape[sample_index]), breaks = seq(0, 1, .01))


```
```{r}
sample_index = num_good_snps_escape >= 5 & est_var_no_escape < .12
sum(sample_index)
hist(unfold(est_skew_no_escape[sample_index]), breaks = seq(0, 1, .01))


```



```{r}
cow_skew_and_stats_df = data.frame(sample_id = wig_samples, 
                                   est_skew = est_skew, est_var = est_var,  #Skew estimates filtering out bad genes
                                   est_skew_total_snps = est_skew_bad_snps, est_var_total_snps = est_var_bad_snps,   #Skew estimates with all genes 
                                   est_skew_no_escape = est_skew_no_escape, est_var_no_escape = est_var_no_escape,   #Skew estimates filtering out escape
                                   num_good_snps = num_good_snps, total_snps = total_snps, num_good_snps_no_escape = num_good_snps_escape)

cow_skew_and_stats_df
```


```{r}

save(cow_skew_and_stats_df, file = '/home/werner/projects/cross_species_XCI/data/species_data_analysis/bos_taurus/cow_skew_and_stats_df_readFilt_diff.Rdata')

```



get the distribution of minimum allele counts for all SNPs in the dataset




```{r}

sample_index = which(num_good_snps_escape >= 10 & est_var_no_escape < .13)

all_min_vec = c()
all_ref_ratios = c()
for(i in 1:length(sample_index)){
  data = filt_cow_vcfs_noEscape[[sample_index[i]]]
  if(is.null(dim(data))){next}
  
  snp_filter = data$Pos %in% snp_stat_df$snp_pos[snp_stat_df$filter]
  
  
  min_sample_vec = vector(mode = 'numeric', length = nrow(data[snp_filter, ]))
  min_index = data$ref_counts[snp_filter] < data$alt_counts[snp_filter]
  min_sample_vec[min_index] = data$ref_counts[snp_filter][min_index]
  min_sample_vec[!min_index] = data$alt_counts[snp_filter][!min_index]
  
  ref_ratios = data$ref_counts[snp_filter] / (data$ref_counts[snp_filter] + data$alt_counts[snp_filter])
  all_ref_ratios = c(all_ref_ratios, ref_ratios)
  all_min_vec = c(all_min_vec, min_sample_vec)
  
}


hist(log10(all_min_vec), breaks = 64, xlab = 'log10 Minimum read counts for all SNPs', main = 'read depth mod')

plot(log10(all_min_vec), all_ref_ratios, xlab = 'log10 Minimum read counts for all SNPs', ylab = 'reference SNP ratio', ylim = c(0,1), cex = .5)

```

```{r}

sample_index = which(num_good_snps_escape >= 10 & est_var_no_escape < .13)

all_min_vec = c()
all_ref_ratios = c()
for(i in 1:length(sample_index)){
  data = filt_cow_vcfs_noEscape[[sample_index[i]]]
  if(is.null(dim(data))){next}
  
  snp_filter = data$Pos %in% snp_stat_df$snp_pos[snp_stat_df$filter]
  
  
  min_sample_vec = vector(mode = 'numeric', length = nrow(data[snp_filter, ]))
  min_index = data$ref_counts[snp_filter] < data$alt_counts[snp_filter]
  min_sample_vec[min_index] = data$ref_counts[snp_filter][min_index]
  min_sample_vec[!min_index] = data$alt_counts[snp_filter][!min_index]
  
  ref_ratios = data$ref_counts[snp_filter] / (data$ref_counts[snp_filter] + data$alt_counts[snp_filter])
  all_ref_ratios = c(all_ref_ratios, ref_ratios)
  all_min_vec = c(all_min_vec, min_sample_vec)
  
}


hist(log10(all_min_vec), breaks = 64, xlab = 'log10 Minimum read counts for all SNPs', main = 'read depth mod')

plot(log10(all_min_vec), all_ref_ratios, xlab = 'log10 Minimum read counts for all SNPs', ylab = 'reference SNP ratio', ylim = c(0,1), cex = .5)

```

```{r}

sample_index = which(num_good_snps_escape >= 10 & est_var_no_escape < .13)

all_min_vec = c()
all_ref_ratios = c()
for(i in 1:length(sample_index)){
  data = filt_cow_vcfs_noEscape[[sample_index[i]]]
  if(is.null(dim(data))){next}
  
  snp_filter = data$Pos %in% snp_stat_df$snp_pos[snp_stat_df$filter]
  
  
  min_sample_vec = vector(mode = 'numeric', length = nrow(data[snp_filter, ]))
  min_index = data$ref_counts[snp_filter] < data$alt_counts[snp_filter]
  min_sample_vec[min_index] = data$ref_counts[snp_filter][min_index]
  min_sample_vec[!min_index] = data$alt_counts[snp_filter][!min_index]
  
  ref_ratios = data$ref_counts[snp_filter] / (data$ref_counts[snp_filter] + data$alt_counts[snp_filter])
  all_ref_ratios = c(all_ref_ratios, ref_ratios)
  all_min_vec = c(all_min_vec, min_sample_vec)
  
}


hist(log10(all_min_vec), breaks = 64, xlab = 'log10 Minimum read counts for all SNPs', main = 'read depth mod')

plot(log10(all_min_vec), all_ref_ratios, xlab = 'log10 Minimum read counts for all SNPs', ylab = 'reference SNP ratio', ylim = c(0,1), cex = .5)

```







compare skew estimates with the different read filtering against the original skew estimates

```{r}
original = load('/home/werner/projects/cross_species_XCI/data/species_data_analysis/bos_taurus/cow_skew_and_stats_df.Rdata')
cow_skew_and_stats_original_df = get(original)

diff_reads = load('/home/werner/projects/cross_species_XCI/data/species_data_analysis/bos_taurus/cow_skew_and_stats_df_readFilt_diff.Rdata')
cow_skew_and_stats_diff_reads_df = get(diff_reads)

```


```{r}
par(pty = 's')
plot(cow_skew_and_stats_original_df$est_skew_no_escape, cow_skew_and_stats_diff_reads_df$est_skew_no_escape, 
     xlab = 'original', ylab = 'read depth mod', main = 'cow estimated skews')
abline(a = 0, b = 1, col = 'red')

par(pty = 's')
plot(cow_skew_and_stats_original_df$num_good_snps_no_escape, cow_skew_and_stats_diff_reads_df$num_good_snps_no_escape, 
     xlab = 'original', ylab = 'read depth mod', main = 'cow # snps')
abline(a = 0, b = 1, col = 'red')



```


```{r}
quantile(cow_skew_and_stats_diff_reads_df$est_var_no_escape, probs = seq(0,1,.1), na.rm = T)
sample_index = cow_skew_and_stats_diff_reads_df$num_good_snps_no_escape >= 10 & cow_skew_and_stats_diff_reads_df$est_var_no_escape < .13 
sum(sample_index)
hist(unfold(cow_skew_and_stats_diff_reads_df$est_skew_no_escape[sample_index]), breaks = seq(0, 1, .01), main = 'read depth mod')


quantile(cow_skew_and_stats_original_df$est_var_no_escape, probs = seq(0,1,.1), na.rm = T)
sample_index = cow_skew_and_stats_original_df$num_good_snps_no_escape >= 10 & cow_skew_and_stats_original_df$est_var_no_escape < .12 
sum(sample_index)
hist(unfold(cow_skew_and_stats_original_df$est_skew_no_escape[sample_index]), breaks = seq(0, 1, .01), main = 'original')



```

```{r}

sample_index = cow_skew_and_stats_diff_reads_df$num_good_snps_no_escape >= 10 & cow_skew_and_stats_diff_reads_df$est_var_no_escape < .13 
var(unfold(cow_skew_and_stats_diff_reads_df$est_skew_no_escape[sample_index]))


sample_index = cow_skew_and_stats_original_df$num_good_snps_no_escape >= 10 & cow_skew_and_stats_original_df$est_var_no_escape < .12 
var(unfold(cow_skew_and_stats_original_df$est_skew_no_escape[sample_index]))



```





#############################################
#############################################


Bootstrapping

Run the bootstrapping_script.R in the command line for parallelization

nohup Rscript bootstrapping_script.R path_to_species_filt_max_vcf path_to_save_boot_results &


```{r}

load('/home/werner/projects/cross_species_XCI/data/species_data_analysis/bos_taurus/bootstrapping_results.Rdata')
length(boots)
boots[1:5]
```



```{r}
lower_CI = vector(mode = 'numeric', length = length(boots))
upper_CI = vector(mode = 'numeric', length = length(boots))

for(i in 1:length(boots)){
  
  if(length(boots[[i]])==1){lower_CI[i] = NA; upper_CI[i] = NA; next} #NA boot
  
  bootstrapped_ests = boots[[i]]$t[ ,1]  #Grab the bootstrapped estimate distribution for the ith sample
  bootstrapped_ests = bootstrapped_ests[order(bootstrapped_ests)]  #Order and get the 95th percentile CI
  lower_CI[i] = bootstrapped_ests[5]
  upper_CI[i] = bootstrapped_ests[195]
}

CI_width = upper_CI - lower_CI



length(CI_width)


```


```{r}
cow_skew_and_stats_df = data.frame(sample_id = wig_samples, 
                                   est_skew = est_skew, est_var = est_var,  #Skew estimates filtering out bad genes
                                   est_skew_total_snps = est_skew_bad_snps, est_var_total_snps = est_var_bad_snps,   #Skew estimates with all genes 
                                   est_skew_no_escape = est_skew_no_escape, est_var_no_escape = est_var_no_escape,   #Skew estimates filtering out escape
                                   num_good_snps = num_good_snps, total_snps = total_snps, num_good_snps_no_escape = num_good_snps_escape,
                                   lower_CI = lower_CI, upper_CI = upper_CI, CI_width = CI_width)

cow_skew_and_stats_df
```



```{r}

save(cow_skew_and_stats_df, file = '/home/werner/projects/cross_species_XCI/data/species_data_analysis/bos_taurus/cow_skew_and_stats_df.Rdata')

```

```{r}

load('/home/werner/projects/cross_species_XCI/data/species_data_analysis/bos_taurus/cow_skew_and_stats_df.Rdata')

```


check out the samples with large variance, large CI, 


```{r}
dim(cow_skew_and_stats_df)

plot(cow_skew_and_stats_df$est_var_no_escape, cow_skew_and_stats_df$CI_width)
quantile(cow_skew_and_stats_df$est_var_no_escape, probs = seq(0,1,.10), na.rm = T)
quantile(cow_skew_and_stats_df$CI_width, probs = seq(0,1,.10), na.rm = T)


```

looking at samples by their variance

```{r}
sample_index = which(cow_skew_and_stats_df$num_good_snps_no_escape >= 30 & cow_skew_and_stats_df$est_var_no_escape >= .12 &
                       cow_skew_and_stats_df$est_skew_no_escape > .6)
length(sample_index)
for(i in 1:length(sample_index)){
  
  data = filt_cow_vcfs_noEscape[[sample_index[i]]]
  ref_ratios = data$ref_counts / (data$ref_counts + data$alt_counts)
  hist(ref_ratios, breaks = seq(0,1,.025))
  
}



```

looking at samples by their confidence interval

```{r}
sample_index = which(cow_skew_and_stats_df$num_good_snps_no_escape >= 10 & cow_skew_and_stats_df$CI_width >= .174 &
                       cow_skew_and_stats_df$est_skew_no_escape > .6)
length(sample_index)
for(i in 1:length(sample_index)){
  
  data = filt_cow_vcfs_noEscape[[sample_index[i]]]
  ref_ratios = data$ref_counts / (data$ref_counts + data$alt_counts)
  hist(ref_ratios, breaks = seq(0,1,.025))
  
}



```


```{r}

sample_index = cow_skew_and_stats_df$num_good_snps_no_escape >= 10 & cow_skew_and_stats_df$CI_width < .174 & cow_skew_and_stats_df$est_var_no_escape < .12
sum(sample_index)
hist(cow_skew_and_stats_df$est_skew_no_escape, breaks = seq(.5,1, .005))
hist(cow_skew_and_stats_df$est_skew_no_escape[sample_index], breaks = seq(.5,1, .005))

```


Get a plot looking the the percent of samples each SNP shows up in the dataset, comparing across species is somewhat of a sense of how inbred the population is? 
Good question for Dan and Adam

```{r}
num_samples = length(filt_cow_vcfs)
snp_stat_df$prop_samples_present = snp_stat_df$num_samples_present / num_samples

snp_filter = snp_stat_df$filter
num_snps_present = sum(snp_filter, na.rm = T)

ggplot(snp_stat_df[snp_filter, ], aes(x = snp_pos, y = prop_samples_present)) + geom_point(alpha = .5, size = .5) + xlab('X chromosome coordinate') + 
  ylab('% of samples SNP is present and passed QC') + ggtitle(sprintf('Het SNPs (%i) present in Bos taurus bulk RNA-seq samples (%i) ',num_snps_present, num_samples))


ggplot(snp_stat_df[snp_filter, ], aes(x = snp_pos, y = mean_total_counts)) + geom_point(alpha = .5, size = .5) + xlab('X chromosome coordinate') + 
  ylab('mean total reads') + ggtitle(sprintf('Het SNPs (%i) present in Bos taurus bulk RNA-seq samples (%i) ',num_snps_present, num_samples))


ggplot(snp_stat_df[snp_filter, ], aes(x = mean_total_counts, y = prop_samples_present)) + geom_point(alpha = .5, size = .5) + xlab('mean read counts per SNP') + 
  ylab('% of samples SNP is present and passed QC') + ggtitle(sprintf('Het SNPs (%i) present in Bos taurus bulk RNA-seq samples (%i) ',num_snps_present, num_samples))



```

```{r}

cow_skew_and_stats_df

```

```{r}

p1 = ggplot(cow_skew_and_stats_df, aes(x = est_skew, y = num_good_snps)) + geom_point(alpha = .5, size = .5) + xlab('Estimated XCI skew') + ylab("Number of SNPs skew estimate is derived from") + ggtitle('Bos taurus samples') + xlim(.5,1)


ggMarginal(p1, type = 'histogram', margin = 'x', breaks = seq(.5,1, .005))

```

```{r}


hist(cow_skew_and_stats_df$CI_width, breaks = 32)

plot(cow_skew_and_stats_df$est_skew, cow_skew_and_stats_df$CI_width)
plot(cow_skew_and_stats_df$num_good_snps, cow_skew_and_stats_df$CI_width)
```




 


























