---
title: "Werner, Hover, and Gillis. Population variability in X-chromosome inactivation across 10 mammalian species. 2024. Code and data for figures."
author: "Jonathan Werner"
date: "2024-09-12"
output: github_document
---



This markdown file contains the code for generating all plots for the publication:
Werner, Hover, and Gillis. Population variability in X-chromosome inactivation across 10 mammalian species. 2024.

All data for these plots is provided at: https://github.com/gillislab/cross_mammal_xci/tree/main/R/data_for_plots



```{r packages}
library(ggplot2)
library(dplyr)
library(ggrepel)
library(ComplexHeatmap)

```


```{r colors}
library(MetBrewer)
species_colors = met.brewer("Tiepolo", n=10,type="continuous")

#Going by the ranking of estimated cell counts
species_palette = c('Macaca' = species_colors[1], 'Rat' = species_colors[2], 'Pig' = species_colors[3],
                    'Goat' = species_colors[4], 'Horse' = species_colors[5], 'Sheep' = species_colors[6],
                    'Human' = species_colors[8], 'Mouse' = species_colors[7], 'Cow' = species_colors[9], 'Dog' = species_colors[10])

species_palette

```




## Figure 1C

```{r species_sample }
#Contains the species_sample_df dataframe
load('/home/werner/projects/cross_species_XCI/final_plots/R/data_for_plots/species_sample_df.Rdata')


filter_label = c(rep('Annotated female', length = 10), rep('>= 10 well-powered SNPs', length = 10))
ggplot(species_sample_df, aes(x = species, y = sample_number, alpha = filter_label, fill = species)) + geom_bar(position="dodge", stat="identity") + coord_flip() +
  ylab('# of bulk RNA-seq samples') +
  scale_fill_manual(values = species_palette, guide = guide_legend(reverse = TRUE)) + scale_alpha_manual(values = c('Annotated female' = 1, '>= 10 well-powered SNPs' = .5)) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
      panel.background = element_rect(fill = 'white'), panel.border = element_rect(color = "black", fill=NA),  
      axis.text.y = element_text(size=12), axis.text.x = element_text(size=12),
      axis.title.y = element_text(size=12), axis.title.x = element_text(size=12)) 

```

## Supp. Figure 1B

```{r num_snps}

load('/home/werner/projects/cross_species_XCI/final_plots/R/data_for_plots/num_snps_df.Rdata')

ggplot(num_snps_df, aes(x = species, y = snp_counts, fill = species)) + geom_violin(scale = 'width') + scale_fill_manual(values = species_palette) +
  ylab('Number of well-powered SNPs per sample')

```


## Supp. Figure 6A

```{r cell_num_error_dot_plots}
#contains the species_squared_error_df dataframe
load(file = '/home/werner/projects/cross_species_XCI/final_plots/R/data_for_plots/species_squared_error_df')


plot(species_squared_error_df$n_vec, species_squared_error_df$horse_error, main = 'Horse squared error', xlab = 'Cell number estimate', ylab = 'Sum squared error')
abline(v = species_squared_error_df$n_vec[which.min(species_squared_error_df$horse_error)], col = 'red')

plot(species_squared_error_df$n_vec, species_squared_error_df$dog_error, main = 'dog squared error', xlab = 'Cell number estimate', ylab = 'Sum squared error')
abline(v = species_squared_error_df$n_vec[which.min(species_squared_error_df$dog_error)], col = 'red')

plot(species_squared_error_df$n_vec, species_squared_error_df$cow_error, main = 'cow squared error', xlab = 'Cell number estimate', ylab = 'Sum squared error')
abline(v = species_squared_error_df$n_vec[which.min(species_squared_error_df$cow_error)], col = 'red')

plot(species_squared_error_df$n_vec, species_squared_error_df$sheep_error, main = 'sheep squared error', xlab = 'Cell number estimate', ylab = 'Sum squared error')
abline(v = species_squared_error_df$n_vec[which.min(species_squared_error_df$sheep_error)], col = 'red')

plot(species_squared_error_df$n_vec, species_squared_error_df$goat_error, main = 'goat squared error', xlab = 'Cell number estimate', ylab = 'Sum squared error')
abline(v = species_squared_error_df$n_vec[which.min(species_squared_error_df$goat_error)], col = 'red')

plot(species_squared_error_df$n_vec, species_squared_error_df$rat_error, main = 'rat squared error', xlab = 'Cell number estimate', ylab = 'Sum squared error')
abline(v = species_squared_error_df$n_vec[which.min(species_squared_error_df$rat_error)], col = 'red')

plot(species_squared_error_df$n_vec, species_squared_error_df$mouse_error, main = 'mouse squared error', xlab = 'Cell number estimate', ylab = 'Sum squared error')
abline(v = species_squared_error_df$n_vec[which.min(species_squared_error_df$mouse_error)], col = 'red')

plot(species_squared_error_df$n_vec, species_squared_error_df$macaca_error, main = 'macaca squared error', xlab = 'Cell number estimate', ylab = 'Sum squared error')
abline(v = species_squared_error_df$n_vec[which.min(species_squared_error_df$macaca_error)], col = 'red')

plot(species_squared_error_df$n_vec, species_squared_error_df$pig_error, main = 'pig squared error', xlab = 'Cell number estimate', ylab = 'Sum squared error')
abline(v = species_squared_error_df$n_vec[which.min(species_squared_error_df$pig_error)], col = 'red')

plot(species_squared_error_df$n_vec, species_squared_error_df$human_error, main = 'human squared error', xlab = 'Cell number estimate', ylab = 'Sum squared error')
abline(v = species_squared_error_df$n_vec[which.min(species_squared_error_df$human_error)], col = 'red')
```


## Figure 2A

```{r XCI_population_distributions}

#Contains the horse_skew_and_stats_df, dog_skew_and_stats_df, cow_skew_and_stats_df, sheep_skew_and_stats_df, goat_skew_and_stats_df, rat_skew_and_stats_df, 
#mouse_skew_and_stats_df, macaca_skew_and_stats_df, pig_skew_and_stats_df, agg_gtex_skew_and_stats_df dataframes
load(file = '/home/werner/projects/cross_species_XCI/final_plots/R/data_for_plots/all_species_skew_and_stats_df.Rdata')

all_skew_and_stats = list(horse_skew_and_stats_df, dog_skew_and_stats_df, cow_skew_and_stats_df, sheep_skew_and_stats_df, 
                          goat_skew_and_stats_df, rat_skew_and_stats_df, mouse_skew_and_stats_df, macaca_skew_and_stats_df, pig_skew_and_stats_df, agg_gtex_skew_and_stats_df)
 
names(all_skew_and_stats) = c('Horse','Dog','Cow','Sheep','Goat','Rat','Mouse','Macaca','Pig', 'Human')


# Functions to fold and unfold ratios 
folded <- function(x) { apply( cbind(x,1-x), 1, max)} 
unfold <- function(x) { c(x,1-x) } 


p = .5
num_bins = 100
binned_x_axis = seq(0,1,1/num_bins)


for(j in 1:length(all_skew_and_stats)){

  fitted_normal_n = species_squared_error_df$n_vec[which.min(species_squared_error_df[ ,j+1])] #first column is the n size vector, skip that
  sigma = sqrt(p*(1-p)/fitted_normal_n)
  dnorm_x = 0:1000/1000
  
  fitted_normal = dnorm(dnorm_x, mean=p, sd = sigma)
  
  theoretical_norms = list()
  n = c(8, 32)
  for(i in 1:length(n)){ 
    sigma_t = sqrt(p*(1-p)/n[i])
    theoretical_norms[[i]] = sigma_t
  }
  
  species_name = names(all_skew_and_stats)[j]
  species_skew_and_stat_df = all_skew_and_stats[[j]]
  specie_color = species_palette[names(species_palette)== species_name]
  
  #Greying out the .4-.6 skews
  fill_vec = rep(specie_color, length(binned_x_axis))
  fill_vec[binned_x_axis >= 0.4 & binned_x_axis <= 0.6] = 'grey'
  color_vec = rep(specie_color, length(binned_x_axis))
  color_vec[binned_x_axis >= 0.4 & binned_x_axis <= 0.6] = 'grey'
  
  
  if(species_name == 'Human'){
    skews = unfold(agg_gtex_skew_and_stats_df$est_skew[agg_gtex_skew_and_stats_df$num_good_snps >= 10])
    df = data.frame(skews = skews)
  }else{
    skews = unfold(filter(species_skew_and_stat_df, num_good_snps_no_escape >= 10 & autosome_imbalance == F)$est_skew_no_escape)
    df = data.frame(skews = skews)
  }
  
  g = ggplot(df, aes(x = skews)) + geom_histogram(aes(y = ..density..), fill = fill_vec, color = color_vec, binwidth = 1/num_bins) +
    xlim(0,1)+ xlab(sprintf('%s XCI ratios', species_name)) + ggtitle(sprintf('%s population XCI ratio distribution', species_name)) +
    stat_function(fun = dnorm, args = list(mean = .5, sd = theoretical_norms[[1]]), n = num_bins, size = .5, alpha = .5, aes(linetype = 'dashed')) +
    stat_function(fun = dnorm, args = list(mean = .5, sd = theoretical_norms[[2]]), n = num_bins, size = .5, alpha = .5, aes(linetype = 'dotted')) + 
    stat_function(fun = dnorm, args = list(mean = .5, sd = sigma), n = num_bins, size = .75, aes(linetype = 'solid')) +
    scale_linetype_manual(name="Estimated cell number",values=c('dashed','dotted','solid'),
                          labels = c('8 cells','32 cells',sprintf('Estimated %i cells', fitted_normal_n)),
                          breaks=c('dashed','dotted','solid'), 
                          guide = guide_legend(override.aes = list(size = c(.5, .5, 2)))) +
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
          panel.background = element_rect(fill = 'white'), panel.border = element_rect(color = "black", fill=NA, size=1),  
          axis.text.y = element_text(size=12), axis.text.x = element_text(size=12),
          axis.title.y = element_text(size=12), axis.title.x = element_text(size=12), 
          legend.position = c(.8,.75))  
  print(g)
  
}


```


## Figure 2B

```{r cell_num_est_confidence}

#Contains the all_ci_df dataframe
load('/home/werner/projects/cross_species_XCI/final_plots/R/data_for_plots/all_ci_df.Rdata')
#Species ordered by evolutionary relationships
all_ci_df$species = factor(all_ci_df$species, levels = c('Dog','Horse','Cow','Goat','Sheep','Pig','Rat','Mouse','Human','Macaca'))
ggplot(all_ci_df, aes(y = species, x = log2(cell_num_est), color = species)) + geom_point(size = 3) + geom_errorbar(aes(xmin = log2(lower_ci), xmax = log2(upper_ci))) +
  scale_color_manual(values = species_palette) + xlim(2, 7) + xlab('Cell divisions (log2 cell # estimate)') + ggtitle('Chromosome X') + ylab('') +
  scale_y_discrete(limits=rev)



```


## Supp. Figure 6 B and C 

```{r}
#Contains the cell_num_stats_df dataframe
load( file = '/home/werner/projects/cross_species_XCI/final_plots/R/data_for_plots/cell_num_stats_df.Rdata')

ori_corr = cor(cell_num_stats_df$estimated_cell_num, cell_num_stats_df$sample_number, method = 'spearman')
ggplot(cell_num_stats_df, aes(y = log10(sample_number), x = estimated_cell_num, color = species)) + geom_point(size = 3) +
  scale_color_manual(values = species_palette) + ylab('log10( Total sample number )') + xlab('Cell # estimate') + ggtitle(sprintf('Correlation: %1.3f', ori_corr))

ori_corr = cor(cell_num_stats_df$estimated_cell_num, cell_num_stats_df$mean_snp, method = 'spearman')
ggplot(cell_num_stats_df, aes(y =  mean_snp, x = estimated_cell_num, color = species)) + geom_point(size = 3) +
  scale_color_manual(values = species_palette)  + ylab('Mean SNP # per sample') + xlab('Cell # estimate') + ggtitle(sprintf('Correlation: %1.3f', ori_corr))

```

## Supp. Figure 6 D

```{r downsampled_cell_num_ests}
#Contains the compare_ci_df dataframe
load( file = '/home/werner/projects/cross_species_XCI/final_plots/R/data_for_plots/compare_ci_df.Rdata')

ggplot(compare_ci_df, aes(y = species, x = log2(cell_num_est), color = color_label, alpha = label)) + geom_point(size = 3) + 
  geom_errorbar(aes(xmin = log2(lower_ci), xmax = log2(upper_ci))) +
  scale_alpha_manual(values = c('Downsampled estimate' = .5, 'Original estimate' = 1)) +
  scale_color_manual(values = species_palette) + xlim(2, 7) + xlab('Cell divisions (log2 cell # estimate)') + ggtitle('Chromosome X') + ylab('') +
  scale_y_discrete(limits=rev)

```





## Figure 3C and Supp. Figure 7 A and B


```{r heterozygosity_and_XCI}
#Contains the cow_het_df, rat_het_df, mouse_het_df, macaca_het_df, pig_het_df, sheep_het_df, goat_het_df, dog_het_df,horse_het_df, human_het_df dataframes
load( '/home/werner/projects/cross_species_XCI/final_plots/R/data_for_plots/all_species_het_dfs.Rdata') 

all_het_dfs = list(cow_het_df, rat_het_df, mouse_het_df, macaca_het_df, pig_het_df, sheep_het_df, goat_het_df, dog_het_df,horse_het_df, human_het_df)
all_species = c('cow','rat','mouse','macaca','pig','sheep','goat','dog','horse', 'human')

for(i in 1:length(all_het_dfs)){

  het_df = all_het_dfs[[i]]
  species_name = all_species[i]
  corr = cor(het_df$het_fraction, het_df$est_skew_no_escape, method = 'spearman', use = 'complete.obs')
  p1 = ggplot(het_df, aes(x = log10(het_fraction), y = est_skew_no_escape)) + geom_bin2d(bins = 25) + scale_fill_continuous(type = "viridis") +
    ylim(.5, 1) + ylab('Estimated XCI ratio') + xlab('Heterozygousity') + ggtitle(sprintf('%s corr: %0.3f', species_name, corr))
  print(p1)
  
  corr = cor(het_df$het_fraction, het_df$est_var_no_escape, method = 'spearman', use = 'complete.obs')
  p2 = ggplot(het_df, aes(x = log10(het_fraction), y = est_var_no_escape)) + geom_bin2d(bins = 25) + scale_fill_continuous(type = "viridis") +
    ylab('Estimated XCI SD') + xlab('Heterozygousity') + ggtitle(sprintf('%s corr: %0.3f',species_name, corr))
  print(p2)

}

```

## Figure 3A

```{r heterozygosity_violins}

species_het_fracs = c(cow_het_df$het_fraction, rat_het_df$het_fraction, mouse_het_df$het_fraction, macaca_het_df$het_fraction, 
                      pig_het_df$het_fraction, sheep_het_df$het_fraction, goat_het_df$het_fraction, dog_het_df$het_fraction,
                      horse_het_df$het_fraction, human_het_df$het_fraction)

species_labels = c(rep('Cow', length = nrow(cow_het_df)), rep('Rat', length = nrow(rat_het_df)), rep('Mouse', length = nrow(mouse_het_df)), 
                   rep('Macaca', length = nrow(macaca_het_df)), 
                   rep('Pig', length = nrow(pig_het_df)), rep('Sheep', length = nrow(sheep_het_df)), rep('Goat', length = nrow(goat_het_df)), 
                   rep('Dog', length = nrow(dog_het_df)), rep('Horse', length = nrow(horse_het_df)),  rep('Human', length = nrow(human_het_df)))

all_species_het_frac_df = data.frame(het_fracs = species_het_fracs, label = species_labels)

median_order_df = all_species_het_frac_df %>% group_by(label) %>% summarize(median = median(het_fracs, na.rm = T)) %>% arrange(desc(median))
all_species_het_frac_df$label = factor( all_species_het_frac_df$label, levels = c(median_order_df$label))

ggplot(all_species_het_frac_df, aes(x = label, y = log10(het_fracs), fill = label)) + geom_violin(scale = 'width') + geom_boxplot(outlier.shape = NA, width = .5, show.legend = F ) +
  ylab('Sample heterozygousity') + xlab('Species') +
  scale_fill_manual(values = species_palette, breaks = levels(all_species_het_frac_df$label))

```


## Figure 3B

```{r heterozygosity_corrs_with_xci}

load( file = '/home/werner/projects/cross_species_XCI/final_plots/R/data_for_plots/het_corr_df_v2.Rdata')

ggplot(het_corr_df, aes(x = corrs_label, y = corrs)) + geom_violin(scale = 'width', fill = 'grey25', alpha = .5) + geom_point(aes(color = species_label), size = 5) +
  scale_color_manual(values = species_palette, breaks = rev(levels(all_species_het_frac_df$label))) +
  scale_fill_manual(values = species_palette, breaks = rev(levels(all_species_het_frac_df$label))) +
  geom_label_repel(aes(label = species_label, fill = species_label), color = 'white',box.padding = 0.5) +
  guides(fill = guide_legend(override.aes = aes(label = ""))) +
  ylab('Correlation with sample heterozygousity (spearman)') + xlab('')

```


## Figure 4 B

```{r XCI_skew_variant_aurocs}

#Contains the snp_skew_auc_df dataframe
load('/home/werner/projects/cross_species_XCI/final_plots/R/data_for_plots/snp_skew_auc_df_4_10_24.Rdata')

species_order_df = snp_skew_auc_df %>% filter(null_label == 'data') %>% group_by(species) %>% summarize(median = median(aucs)) %>% arrange(desc(median))
ggplot(filter(snp_skew_auc_df, reference_bias_index == FALSE), aes(x = species, y = aucs, color = null_label, fill = species, alpha = null_label)) + 
  geom_violin(scale = 'width', width = .75) +
  scale_fill_manual(values = species_palette, breaks = levels(snp_skew_auc_df$species)) +
  scale_color_manual(values = c('null' = 'grey', 'data' = 'black')) +
  scale_alpha_manual(values = c('null' = .5, 'data' = 1)) +
  ylim(0, 1) +
  geom_hline(yintercept = .5, color = 'red', linetype = 'dashed')

```
## Figure 4C

```{r variant_freq_and_xci_auroc}
#contains the snp_skew_auc_with_meta_df dataframe
load('/home/werner/projects/cross_species_XCI/final_plots/R/data_for_plots/snp_skew_auc_with_meta_4_10_24.Rdata')


ggplot(snp_skew_auc_with_meta_df, aes(y = variant_prevalence, x = skew_auc)) + geom_point(size = 1, alpha = .25) + 
  geom_vline(xintercept = .5, linetype = 'dashed', color = 'red') +
  scale_color_manual(values = species_palette) +
  xlab('Strength of XCI ratio and variant association (AUROC)') + ylab('Variant prevalence')



```


## Figure 4D

```{r species_zoom}

temp_skew_snp_df = snp_skew_auc_with_meta_df %>% filter(skew_auc >= .75 & skew_auc_significance == TRUE & label != 'Human')
temp_skew_snp_df = temp_skew_snp_df %>% group_by(label, gene) %>% mutate(gene = replace(gene,  expression < max(expression), ""))
#Keep the full annotated gene list to include in the paper text
temp_skew_snp_df

ggplot(temp_skew_snp_df, aes(x = skew_auc, y = variant_prevalence, color = label)) + geom_point(size = 2,show.legend = F) +
  scale_color_manual(values = species_palette) + facet_wrap(~label,ncol = 3, scales = 'free_y') + geom_label_repel(aes(label = gene), size = 3, force = 3, show.legend = F, max.overlaps = 15) +
  xlab('Strength of XCI ratio and variant association (AUROC)') + ylab('Variant prevalence') 


```


## Supp. Figure 8 A, B, C, and D

```{r supp_fig_8_scatters}

ggplot(snp_skew_auc_with_meta_df, aes(y = variant_prevalence, x = skew_auc, color = skew_auc_significance)) + geom_point(size = 1, alpha = .25) + 
  geom_vline(xintercept = .75, linetype = 'dashed', color = 'red') +
  scale_color_manual(values = c('TRUE' = 'red', 'FALSE' = 'black'), name = 'AUROC significance') +
  xlab('Strength of XCI ratio and variant association (AUROC)') + ylab('Variant prevalence')

ggplot(snp_skew_auc_with_meta_df, aes(y = skew_auc_power_significance, x = skew_auc, color = skew_auc_significance)) + geom_point(size = 1, alpha = .25) + 
  scale_color_manual(values = c('TRUE' = 'red', 'FALSE' = 'black'), name = 'AUROC significance') +
  xlab('AUROC') + ylab('Estimated AUROC power') +
  geom_vline(xintercept = .75, linetype = 'dashed', color = 'red')

ggplot(snp_skew_auc_with_meta_df, aes(y = skew_auc_power_effect_75, x = skew_auc, color = skew_auc_significance)) + geom_point(size = 1, alpha = .25) + 
  scale_color_manual(values = c('TRUE' = 'red', 'FALSE' = 'black'), name = 'AUROC significance') +
  geom_vline(xintercept = .75, linetype = 'dashed', color = 'red') +
  xlab('AUROC') + ylab('Estimated AUROC power effect size .75')

ggplot(snp_skew_auc_with_meta_df, aes(y = skew_auc_bootstrap_var, x = skew_auc, color = skew_auc_significance)) + geom_point(size = 1, alpha = .25) + 
  scale_color_manual(values = c('TRUE' = 'red', 'FALSE' = 'black'), name = 'AUROC significance') +
  geom_vline(xintercept = .75, linetype = 'dashed', color = 'red') +
  xlab('AUROC') + ylab('Variance of bootstrapped AUROC distribution')

```
## Supp. Figure 8E

```{r ref_check_variants}
#Contains the human_ref_check_df, cow_ref_check_df, pig_ref_check_df, sheep_ref_check_df, goat_ref_check_df, horse_ref_check_df, 
#rat_ref_check_df, mouse_ref_check_df, macaca_ref_check_df, dog_ref_check_df dataframes
load(file = '/home/werner/projects/cross_species_XCI/final_plots/R/data_for_plots/all_ref_check_dfs.Rdata')


ggplot(human_ref_check_df, aes(x = ref_biased_index, y = auc, fill = ref_biased_index)) + geom_violin(scale = 'width') + geom_boxplot(width = .25) + ggtitle('human') +
  xlab('Reference biased SNPs') + ylab('XCI ratio and variant AUROC') 

ggplot(cow_ref_check_df, aes(x = ref_biased_index, y = auc, fill = ref_biased_index)) + geom_violin(scale = 'width') + geom_boxplot(width = .25) + ggtitle('cow') +
  xlab('Reference biased SNPs') + ylab('XCI ratio and variant AUROC') 

ggplot(pig_ref_check_df, aes(x = ref_biased_index, y = auc, fill = ref_biased_index)) + geom_violin(scale = 'width') + geom_boxplot(width = .25) + ggtitle('pig') +
  xlab('Reference biased SNPs') + ylab('XCI ratio and variant AUROC') 

ggplot(sheep_ref_check_df, aes(x = ref_biased_index, y = auc, fill = ref_biased_index)) + geom_violin(scale = 'width') + geom_boxplot(width = .25) + ggtitle('sheep') +
  xlab('Reference biased SNPs') + ylab('XCI ratio and variant AUROC') 

ggplot(goat_ref_check_df, aes(x = ref_biased_index, y = auc, fill = ref_biased_index)) + geom_violin(scale = 'width') + geom_boxplot(width = .25) + ggtitle('goat') +
  xlab('Reference biased SNPs') + ylab('XCI ratio and variant AUROC') 

ggplot(horse_ref_check_df, aes(x = ref_biased_index, y = auc, fill = ref_biased_index)) + geom_violin(scale = 'width') + geom_boxplot(width = .25) + ggtitle('horse') +
  xlab('Reference biased SNPs') + ylab('XCI ratio and variant AUROC') 

ggplot(rat_ref_check_df, aes(x = ref_biased_index, y = auc, fill = ref_biased_index)) + geom_violin(scale = 'width') + geom_boxplot(width = .25) + ggtitle('rat') +
  xlab('Reference biased SNPs') + ylab('XCI ratio and variant AUROC') 

ggplot(mouse_ref_check_df, aes(x = ref_biased_index, y = auc, fill = ref_biased_index)) + geom_violin(scale = 'width') + geom_boxplot(width = .25) + ggtitle('mouse') +
  xlab('Reference biased SNPs') + ylab('XCI ratio and variant AUROC') 

ggplot(macaca_ref_check_df, aes(x = ref_biased_index, y = auc, fill = ref_biased_index)) + geom_violin(scale = 'width') + geom_boxplot(width = .25) + ggtitle('macaca') +
  xlab('Reference biased SNPs') + ylab('XCI ratio and variant AUROC') 

ggplot(dog_ref_check_df, aes(x = ref_biased_index, y = auc, fill = ref_biased_index)) + geom_violin(scale = 'width') + geom_boxplot(width = .25) + ggtitle('dog') +
  xlab('Reference biased SNPs') + ylab('XCI ratio and variant AUROC') 


```



## Supp. Figure 9B

```{r cow_down_sample_variant_auc}
#Contains the all_cow_downsample_df dataframe
load('/home/werner/projects/cross_species_XCI/final_plots/R/data_for_plots/all_cow_downsample_df.Rdata')

ori_variant_data = filter(snp_skew_auc_with_meta_df, label == 'Cow' & gene == 'KDM6A')
downSamp_variant_data = filter(all_cow_downsample_df, gene == 'KDM6A') %>% group_by(variants) %>% summarize(mean_auc = mean(auc), max_auc = max(auc))

index = match(ori_variant_data$variants, downSamp_variant_data$variants)
downSamp_variant_data = downSamp_variant_data[index, ]
downSamp_variant_data[ ,c(2,3)][is.na(downSamp_variant_data[ ,c(2,3)])] = 0

par(pty = 's')
plot(ori_variant_data$skew_auc, downSamp_variant_data$mean_auc, ylim = c(0, 1), xlim = c(0, 1), ylab = "Mean AUC of downsampled data", xlab='Original AUC of variant',
     main = 'Cow: KDM6A' )
abline(v = .75, col = 'red')
abline(h = .75, col = 'red')



ori_variant_data = filter(snp_skew_auc_with_meta_df, label == 'Cow' & gene == 'TMEM164')
downSamp_variant_data = filter(all_cow_downsample_df, gene == 'TMEM164') %>% group_by(variants) %>% summarize(mean_auc = mean(auc), max_auc = max(auc))

index = match(ori_variant_data$variants, downSamp_variant_data$variants)
downSamp_variant_data = downSamp_variant_data[index, ]
downSamp_variant_data[ ,c(2,3)][is.na(downSamp_variant_data[ ,c(2,3)])] = 0

par(pty = 's')
plot(ori_variant_data$skew_auc, downSamp_variant_data$mean_auc, ylim = c(0, 1), xlim = c(0, 1), ylab = "Mean AUC of downsampled data", xlab='Original AUC of variant',
     main = 'Cow: TMEM164' )
abline(v = .75, col = 'red')
abline(h = .75, col = 'red')



ori_variant_data = filter(snp_skew_auc_with_meta_df, label == 'Cow' & gene == 'MAMLD1')
downSamp_variant_data = filter(all_cow_downsample_df, gene == 'MAMLD1') %>% group_by(variants) %>% summarize(mean_auc = mean(auc), max_auc = max(auc))

index = match(ori_variant_data$variants, downSamp_variant_data$variants)
downSamp_variant_data = downSamp_variant_data[index, ]
downSamp_variant_data[ ,c(2,3)][is.na(downSamp_variant_data[ ,c(2,3)])] = 0

par(pty = 's')
plot(ori_variant_data$skew_auc, downSamp_variant_data$mean_auc, ylim = c(0, 1), xlim = c(0, 1), ylab = "Mean AUC of downsampled data", xlab='Original AUC of variant',
     main = 'Cow: MAMLD1' )
abline(v = .75, col = 'red')
abline(h = .75, col = 'red')



ori_variant_data = filter(snp_skew_auc_with_meta_df, label == 'Cow' & gene == 'MID2')
downSamp_variant_data = filter(all_cow_downsample_df, gene == 'MID2') %>% group_by(variants) %>% summarize(mean_auc = mean(auc), max_auc = max(auc))

index = match(ori_variant_data$variants, downSamp_variant_data$variants)
downSamp_variant_data = downSamp_variant_data[index, ]
downSamp_variant_data[ ,c(2,3)][is.na(downSamp_variant_data[ ,c(2,3)])] = 0

par(pty = 's')
plot(ori_variant_data$skew_auc, downSamp_variant_data$mean_auc, ylim = c(0, 1), xlim = c(0, 1), ylab = "Mean AUC of downsampled data", xlab='Original AUC of variant',
     main = 'Cow: MID2' )
abline(v = .75, col = 'red')
abline(h = .75, col = 'red')

```

## Supp. Figure 9 C

```{r individual_variant_per_species}

#Contains the sheep comb_mxra5_df dataframe
load(file = '/home/werner/projects/cross_species_XCI/final_plots/R/data_for_plots/sheep_individual_variant_auc_boxplots.Rdata')

ggplot(comb_mxra5_df, aes(y = skews, x = variant, fill = data_label)) + geom_boxplot() + xlab('sheep MXRA5') + ylab('XCI skews') + ggtitle('sheep MXRA5')



#Contains the cow comb_mamld1_df, comb_mid2_df, comb_tmem164_df, comb_kdm6a_df dataframes
load(file = '/home/werner/projects/cross_species_XCI/final_plots/R/data_for_plots/cow_individual_variant_auc_boxplots.Rdata')

ggplot(comb_mamld1_df, aes(y = skews, x = variant, fill = data_label)) + geom_boxplot() + xlab('cow MAMLD1') + ylab('XCI skews') + ggtitle('cow MAMLD1')

ggplot(comb_mid2_df, aes(y = skews, x = variant, fill = data_label)) + geom_boxplot() + xlab('cow MID2') + ylab('XCI skews') + ggtitle('cow MID2')

ggplot(comb_tmem164_df, aes(y = skews, x = variant, fill = data_label)) + geom_boxplot() + xlab('cow TMEM164') + ylab('XCI skews') + ggtitle('cow TMEM164')

ggplot(comb_kdm6a_df, aes(y = skews, x = variant, fill = data_label)) + geom_boxplot() + xlab('cow KDM6A') + ylab('XCI skews') + ggtitle('cow KDM6A')



#Contains the goat comb_msl3_df, comb_mxra5_df, comb_shroom2_df, comb_loc102_df dataframes
load(file = '/home/werner/projects/cross_species_XCI/final_plots/R/data_for_plots/goat_individual_variant_auc_boxplots.Rdata')

ggplot(comb_msl3_df, aes(y = skews, x = variant, fill = data_label)) + geom_boxplot() + xlab('goat MSL3') + ylab('XCI skews') + ggtitle('goat MSL3')

ggplot(comb_mxra5_df, aes(y = skews, x = variant, fill = data_label)) + geom_boxplot() + xlab('goat MXRA5') + ylab('XCI skews') + ggtitle('goat MXRA5')

ggplot(comb_shroom2_df, aes(y = skews, x = variant, fill = data_label)) + geom_boxplot() + xlab('goat SHROOM2') + ylab('XCI skews') + ggtitle('goat SHROOM2')

ggplot(comb_loc102_df, aes(y = skews, x = variant, fill = data_label)) + geom_boxplot() + xlab('goat LOC102168615') + ylab('XCI skews') + ggtitle('goat LOC102168615')



#Contains the horse comb_arse_df, comb_plcxd1_df, comb_gpc4_df, comb_arhgap6_df dataframes
load(file = '/home/werner/projects/cross_species_XCI/final_plots/R/data_for_plots/horse_individual_variant_auc_boxplots.Rdata')

ggplot(comb_arhgap6_df, aes(y = skews, x = variant, fill = data_label)) + geom_boxplot() + xlab('Horse ARHGAP6') + ylab('XCI skews') + ggtitle('Horse ARHGAP6')

ggplot(comb_gpc4_df, aes(y = skews, x = variant, fill = data_label)) + geom_boxplot() + xlab('Horse GPC4') + ylab('XCI skews') + ggtitle('Horse GPC4')

ggplot(comb_plcxd1_df, aes(y = skews, x = variant, fill = data_label)) + geom_boxplot() + xlab('Horse PLCXD1') + ylab('XCI skews') + ggtitle('Horse PLCXD1')

ggplot(comb_arse_df, aes(y = skews, x = variant, fill = data_label)) + geom_boxplot() + xlab('Horse ARSE') + ylab('XCI skews') + ggtitle('Horse ARSE')


```









## Supp. Figure 10 A and B

```{r mouse_haplotypes}
#Contains the mouse_xist_present_mat and mouse_hap_df objects
load(file = '/home/werner/projects/cross_species_XCI/final_plots/R/data_for_plots/mouse_haplotype_data.Rdata' )

col_annot = HeatmapAnnotation(study = mouse_hap_df$study,
                              col = list(study = c('Study_1_striatum' = '#F8766D', 'Study_2_pancreas' = '#00BFC4')))
colors = structure(1:2, names = c(0, 1)) # black, red
h1 = Heatmap(mouse_xist_present_mat, name = "mat", col = colors,top_annotation = col_annot,
        clustering_distance_columns = function(m) dist(m, method = 'binary'),
        clustering_distance_rows = function(m) dist(m, method = 'binary'),
        show_column_names = F, show_row_names = F,
        clustering_method_rows = "ward.D2", clustering_method_columns = "ward.D2", show_column_dend = T, column_split = 4)

h1 = draw(h1)



mouse_hap_df$label = rep('other', length = nrow(mouse_hap_df))
mouse_hap_df$label[column_order(h1)[[1]]] = 'haplotype_group 1'
mouse_hap_df$label[column_order(h1)[[2]]] = 'haplotype_group 2'
mouse_hap_df$label[column_order(h1)[[3]]] = 'haplotype_group 3'
mouse_hap_df$label[column_order(h1)[[4]]] = 'haplotype_group 4'

ggplot(mouse_hap_df, aes(x = study, y = est_skew_no_escape, color = label)) + geom_boxplot() + xlab('Mouse XIST putative haplotypes') + ylab('Estimated XCI ratio') +
  geom_point(position=position_dodge(width=0.75),aes(group=label), color = 'grey50')


ggplot(mouse_hap_df, aes(x = label, y = est_skew_no_escape, color = study)) + geom_boxplot() + xlab('Mouse XIST putative haplotypes') + ylab('Estimated XCI ratio') +
  scale_x_discrete(guide = guide_axis(n.dodge=2)) +
  geom_point(position=position_dodge(width=0.75),aes(group=study), color = 'grey50')


```



## Supp. Figure 3A
This plot is only showing the confirmed female samples that had y-chromosome CPM <200. We had deleted the male samples before I saved the data for the plot.
```{r mouse_sex_classification}
#Contains the mouse_sex_df dataframe
load(file = '/home/werner/projects/cross_species_XCI/final_plots/R/data_for_plots/mouse_sex_df.Rdata')

ggplot(mouse_sex_df, aes(x = x_cpm_counts, y = y_cpm_counts, color = seq_type)) + geom_point() + ylab('agg. Y-chromosome CPM counts') + xlab('agg. X-chromosome CPM counts') + 
  ggtitle('Mouse sex chromosome expression')

```

















